subprojects/core/src/main/java/org/gradle/api/internal/tasks/userinput/UserQuestions.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.userinput;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+
+public interface UserQuestions {
+    /**
+     * Prompts the user with a yes/no question and returns the answer. Requires that the user explicitly type "yes" or "no".
+     *
+     * @param question The text of the question.
+     * @return the answer or {@code null} if not connected to a console.
```

Comment:
❓ This `null` requirement seems really out of place here.  Why does this method not provide a default like the others?

Reply:
It's because the method is used when the caller would like the user to explicitly type "yes" or "no", for example when accepting a license.

We could probably change this to return something more strongly typed, but in this PR I didn't want to change the APIs for the specific questions, just rework the overall structure. We can improve the questions in a later PR.